src/index.js

```diff
@@ -330,11 +432,21 @@ class PublicationClient extends EventEmitter {
    *    collection to add it to.
    */
   _onAdded(message) {
-    var collectionName = message.collection;
-    var id = message.id;
-    var fields = message.fields;
+    const collectionName = message.collection;
+    const id = message.id;
+    const fields = message.fields;
+
+    if (this.isCollectionSubscribedViaCallbacks(collectionName)) {
+      if (this._subscribedToCallbacks) {
+        this._onAddedEvent(collectionName, id, fields);
+      }
+      // check if backbone collection also listens for the publication
+      if (!this.isCollectionCreated(collectionName)) {
+        return;
+      }
+    }
 
```

Comment:
@dnechay I think this is it :D 

the 439-440 is double check, you shouldn't have leftovers in the subscription queue that were subscribed to callbacks.

When subscribing like this I also add a special parameter in the key to differentiate between my events and the rests. 

```
    return !!_.find(
      _.keys(this._subscriptions),
      (key) => key.startsWith(`["${collection}"`) && key.indexOf('subscribedViaCallbacks') > -1
    );
```
I'm looking for startsWith because sometimes you can have the collection name in the options (I saw it in `feature` it was the key with `sequences` in the options).

The logic is like this:

Does the collection was subscribed by callbacks?
- no, follow the normal flow
- yes
      - call the callback
      - check if collection is already subscribed to backbone publications then do the normal flow
         

<img width="618" alt="Screenshot 2022-08-10 at 15 34 23" src="https://user-images.githubusercontent.com/7041061/183902472-c18a93ab-7865-4199-9007-32d61841bba6.png">

Reply:
cc @bradvogel

Reply:
Thanks for the deep explanation @PandasProperty! ❤️‍🔥 
I was going to ask why then not using normal flow also in redux, but see you already done it 👍🏼